completions/tipc

```diff
@@ -0,0 +1,306 @@
+    link)
+        let optind++
+
+        case "${words[$optind]}" in
+        get)
+            let optind++
+
+            if [[ $cword -eq $optind ]]; then
+                _tipc_link_opts
+            elif [[ $cword -ge $optind+1 ]]; then
+                _tipc_link $(($optind + 1)) "peers"
+            fi
+            return 0
+            ;;
+        set)
+            let optind++
+
+            if [[ $cword -eq $optind ]]; then
+                _tipc_link_opts
+            elif [[ $cword -ge $optind+2 ]]; then
+                _tipc_link $(($optind + 2)) "peers"
+            fi
+            return 0
+            ;;
+        list)
+            return 0
+            ;;
+        statistics)
+            let optind++
+
+            case "${words[$optind]}" in
+            show|reset)
+                _tipc_link $(($optind + 1))
+                return 0
+                ;;
+            esac
+
+            COMPREPLY=( $( compgen -W 'show reset' -- $cur) )
+            return 0
+            ;;
+        esac
+
+        COMPREPLY=( $( compgen -W 'get set list statistics'  -- $cur ) )
+        return 0
+        ;;
+    media)
+        let optind++
+
+        case "${words[$optind]}" in
+        get)
+            let optind++
+
+            if [[ $cword -eq $optind ]]; then
+                _tipc_link_opts
+            elif [[ $cword -ge $optind+1 ]]; then
+                _tipc_media $(($optind + 1))
+            fi
+            return 0
+            ;;
+        set)
+            let optind++
+
+            if [[ $cword -eq $optind ]]; then
+                _tipc_link_opts
+            elif [[ $cword -ge $optind+2 ]]; then
+                _tipc_media $(($optind + 2))
+            fi
+            return 0
+            ;;
+        list)
+            return 0
+            ;;
+        esac
+
+        COMPREPLY=( $( compgen -W 'get set list'  -- $cur ) )
+        return 0
+        ;;
+    nametable)
+        let optind++
+
+        case "${words[$optind]}" in
+        show)
+            return 0
+            ;;
+        esac
+
+        COMPREPLY=( $( compgen -W 'show'  -- $cur ) )
+        return 0
+        ;;
+    node)
+        let optind++
+
+        case "${words[$optind]}" in
+        get|set)
+            let optind++
+
+            case "${words[$optind]}" in
+            address|netid)
+                return 0
+                ;;
+            esac
+            COMPREPLY=( $( compgen -W 'address netid'  -- $cur ) )
+            return 0
+            ;;
+        list)
+            return 0
+            ;;
+        esac
+
+        COMPREPLY=( $( compgen -W 'list get set'  -- $cur ) )
+        return 0
+        ;;
+    socket)
+        let optind++
+
+        case "${words[$optind]}" in
+        list)
+            return 0
+            ;;
+        esac
+
+        COMPREPLY=( $( compgen -W 'list'  -- $cur ) )
+        return 0
+        ;;
+    esac
+
+    COMPREPLY=( $( compgen -W 'bearer link media nametable node socket' -- $cur ) )
+}
+
```

Comment:
Looks like a bit too much was removed from here, something like `&& complete -F _tipc tipc` is still needed (just the `$have` part was supposed to go away).